solidity/contracts/BancorUpgradeConverter.sol

```diff
@@ -0,0 +1,236 @@
+    // triggered when the upgrading process is done
+    event ConverterCreated(address indexed _fromConverter, address indexed _toConverter);
+
+    /**
+        @dev constructor
+    */
+    function BancorUpgradeConverter()
+        public
+    {}
+
+    /**
+        @dev upgrade from old converter versions to the latest version
+        will throw if transfer ownership request was not sent before calling this function.
+        ownership of the current and new converter will be transfered to a given owner address.
+        when process is finish the function fired an event.
+
+        @param _fromConverter   converter to upgrade
+        @param _version         current conveter version
+    */
+    function upgradeConverter(IBancorConverter _fromConverter, bytes32 _version)
+        public
+    {
+        bool formerVersions = false;
+        if (_version == "0.4")
+            formerVersions = true;
+        acceptConverterOwnership(_fromConverter);
+        IBancorConverter toConverter = createConverter(_fromConverter);
+        copyConnectors(_fromConverter, toConverter, formerVersions);
+        copyConvertionFee(_fromConverter, toConverter);
+        copyQuickBuyPath(_fromConverter, toConverter);
+        transferConnectorsBalances(_fromConverter, toConverter, formerVersions);
+        _fromConverter.transferTokenOwnership(toConverter);
+        toConverter.acceptTokenOwnership();
+        _fromConverter.transferOwnership(msg.sender);
+        toConverter.transferOwnership(msg.sender);
+        toConverter.transferManagement(msg.sender);
+        ConverterCreated(address(_fromConverter), address(toConverter));
+    }
+
+    /**
+        @dev first step when upgrading a converter is to transfer the ownership to the upgrade converter
+        contract instance. since transferring ownership has two steps, the instance must to accept the
+        ownership before it can proceed the upgrading process.
+        notice that even that the ownership transfered to the contract, only the current owner can 
+        execute the upgrade process.
+        when the process is finish, the function fired an event.
+
+        @param _fromConverter       converter to accept its ownership request
+
+        @return the new converter
+    */
+    function acceptConverterOwnership(IBancorConverter _fromConverter) private {
+        _fromConverter.acceptOwnership();
+        ConverterOwned(_fromConverter, this);
+    }
+
+    /**
+        @dev create a new converter with same token and the same max conversion fee and the same 
+        extensions address as set in the given converter.
+        the new converter has no connectors.
+
+        @param _fromConverter       converter to read from
+
+        @return the new converter
+    */
+    function createConverter(IBancorConverter _fromConverter) private returns(IBancorConverter) {
+        IERC20Token emptyConnector = IERC20Token(address(0));
```

Comment:
seems unnecessary as it's an optional argument, can't just pass `address(0)` to the constructor?

Reply:
fixed